cluster/swarm/cluster.go

```diff
@@ -151,32 +151,45 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 	}
 
 	nodes, err := c.scheduler.SelectNodesForContainer(c.listNodes(), configTemp)
+	c.scheduler.Unlock()
+
 	if err != nil {
-		c.scheduler.Unlock()
 		return nil, err
 	}
-	n := nodes[0]
-	engine, ok := c.engines[n.ID]
-	if !ok {
+
+	for _, n := range nodes {
```

Comment:
There should be a way to limit this I think - we might end up retrying on every node on the cluster, not sure this is what we want